EmberBSD local, AI-assisted; not submitted upstream.
Update MPC to current releases; preserve and refresh pkgsrc portability
patches. Source URLs, hashes, upstream patch removals and ABI notes are in
../prerequisites.md. Base pkgsrc revision remains unchanged.

diff -ruN a/math/mpfr/Makefile b/math/mpfr/Makefile
--- a/math/mpfr/Makefile
+++ b/math/mpfr/Makefile
@@ -1,8 +1,8 @@
 # $NetBSD: Makefile,v 1.48 2023/09/09 20:15:54 wiz Exp $
 
-DISTNAME=	mpfr-4.2.1
+DISTNAME=	mpfr-4.2.2
 CATEGORIES=	math
-MASTER_SITES=	http://www.mpfr.org/${DISTNAME}/
+MASTER_SITES=	https://www.mpfr.org/${DISTNAME}/
 EXTRACT_SUFX=	.tar.bz2
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
diff -ruN a/math/mpfr/distinfo b/math/mpfr/distinfo
--- a/math/mpfr/distinfo
+++ b/math/mpfr/distinfo
@@ -1,5 +1,5 @@
 $NetBSD: distinfo,v 1.35 2023/09/09 20:15:54 wiz Exp $
 
-BLAKE2s (mpfr-4.2.1.tar.bz2) = 3d4adcc8e69386a962b2350a456a27a44d693fbe603cbbc954725887d88febc2
-SHA512 (mpfr-4.2.1.tar.bz2) = c81842532ecc663348deb7400d911ad71933d3b525a2f9e5adcd04265c9c0fdd1f22eca229f482703ac7f222ef209fc9e339dd1fa47d72ae57f7f70b2336a76f
-Size (mpfr-4.2.1.tar.bz2) = 1720899 bytes
+BLAKE2s (mpfr-4.2.2.tar.bz2) = 2b507bb26e95f69ed5df2fc1b5b07eb56f9d7bbdc7a86c9488246a01f586b211
+SHA512 (mpfr-4.2.2.tar.bz2) = 0176e50808dcc07afbf5bc3e38bf9b7b21918e5f194aa0bfd860d99b00c470630aef149776c4be814a61c44269c3a5b9a4b0b1c0fcd4c9feb1459d8466452da8
+Size (mpfr-4.2.2.tar.bz2) = 1753999 bytes
diff -ruN a/math/mpcomplex/Makefile b/math/mpcomplex/Makefile
--- a/math/mpcomplex/Makefile	2026-10-02 21:24:59.000000000 +0300
+++ b/math/mpcomplex/Makefile	2026-10-06 23:01:43.371112394 +0300
@@ -1,7 +1,8 @@
 # $NetBSD: Makefile,v 1.14 2025/12/23 07:48:21 wiz Exp $
 
-DISTNAME=	mpc-1.3.1
+DISTNAME=	mpc-1.4.1
 PKGNAME=	${DISTNAME:S/mpc/mpcomplex/}
+EXTRACT_SUFX=	.tar.xz
 CATEGORIES=	math
 MASTER_SITES=	${MASTER_SITE_GNU:=mpc/}
 
diff -ruN a/math/mpcomplex/PLIST b/math/mpcomplex/PLIST
--- a/math/mpcomplex/PLIST	2026-10-02 21:24:59.000000000 +0300
+++ b/math/mpcomplex/PLIST	2026-10-06 23:01:43.371321927 +0300
@@ -2,3 +2,4 @@
 include/mpc.h
 info/mpc.info
 lib/libmpc.la
+lib/pkgconfig/mpc.pc
diff -ruN a/math/mpcomplex/distinfo b/math/mpcomplex/distinfo
--- a/math/mpcomplex/distinfo	2026-10-02 21:24:59.000000000 +0300
+++ b/math/mpcomplex/distinfo	2026-10-06 23:10:44.352650481 +0300
@@ -1,6 +1,6 @@
 $NetBSD: distinfo,v 1.12 2023/10/29 18:06:32 wiz Exp $
 
-BLAKE2s (mpc-1.3.1.tar.gz) = 5d2921ff2ec9523efd7063c63ee57dcb0905b398a8a2a2e58a952846e4dc934b
-SHA512 (mpc-1.3.1.tar.gz) = 4bab4ef6076f8c5dfdc99d810b51108ced61ea2942ba0c1c932d624360a5473df20d32b300fc76f2ba4aa2a97e1f275c9fd494a1ba9f07c4cb2ad7ceaeb1ae97
-Size (mpc-1.3.1.tar.gz) = 773573 bytes
-SHA1 (patch-src_get__x.c) = 25495981765971c031b5a92dce098a44e53413c8
+BLAKE2s (mpc-1.4.1.tar.xz) = ec9bf1d17fe24b9877c80724c1ea20ad102359783338c8836132feb7dcaee86e
+SHA512 (mpc-1.4.1.tar.xz) = cdfbe64e80e30df64dbd536e8104a6bb10471582415d5b45a1dd30eb2a312781b38389a9531243413c14e0f88866a35159b20e77900718f28c1030253a1305b9
+Size (mpc-1.4.1.tar.xz) = 531992 bytes
+SHA1 (patch-src_get__x.c) = 7f57257e355c1577edd2c88f9edcf713188ae668
diff -ruN a/math/mpcomplex/patches/patch-src_get__x.c b/math/mpcomplex/patches/patch-src_get__x.c
--- a/math/mpcomplex/patches/patch-src_get__x.c	2026-10-02 21:24:59.000000000 +0300
+++ b/math/mpcomplex/patches/patch-src_get__x.c	2026-10-06 23:01:43.371512162 +0300
@@ -4,10 +4,10 @@
 
 --- src/get_x.c.orig	2011-11-04 18:20:19.000000000 +0000
 +++ src/get_x.c
-@@ -35,6 +35,13 @@ along with this program. If not, see htt
+@@ -28,6 +28,13 @@
  #include "mpc-impl.h"
  
- #ifdef HAVE_COMPLEX_H
+ #ifdef _MPC_HAVE_COMPLEX_H
 +
 +#if defined(__sun) && defined(__GNUC__) && defined(_Imaginary_I)
 +#undef I
@@ -15,6 +15,6 @@
 +
 +#endif
 +
- double _Complex
+ DOUBLE_COMPLEX
  mpc_get_dc (mpc_srcptr op, mpc_rnd_t rnd) {
     return I * mpfr_get_d (mpc_imagref (op), MPC_RND_IM (rnd))
